variables.go

```diff
@@ -228,7 +181,7 @@ func (e *Executor) compiledTask(call *ast.Call, evaluateShVars bool) (*ast.Task,
 		}
 	}
 
-	if len(origTask.Status) > 0 {
```

Comment:
I'm not a big fan of those two nested if statements.
`TIMESTAMP` and `CHECKSUM` should only be available if there are sources defined. I'd replace `len(origTask.Status) > 0` with `len(origTask.Source) > 0` and move it above `cmds`.

Reply:
True, good idea. I changed it like you said but now it also has the effect that .TIMESTAMP and .CHECKSUM are available in preconditions and deps. Do you want to keep it like that?

Reply:
Indeed this is wanted :slightly_smiling_face: